### PR TITLE
Q285 Add inorderSuccessorInBST.py

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: inorderSuccessorInBST.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        """
+        :type root: TreeNode
+        :type p: TreeNode
+        :rtype: TreeNode
+        """
+        if not root:
+            return None
+        if root.val < p.val:
+            return self.inorderSuccessor(root.right, p)
+        elif root.val == p.val:
```
Comment (nkukarl) [inline inorderSuccessorInBST.py]: `elif` -> `if`

File: inorderSuccessorInBST.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            return self.findSmallest(root.right)
+        elif self.findLargest(root.left) == p:
```
Comment (nkukarl) [inline inorderSuccessorInBST.py]: `elif` -> `if`

File: inorderSuccessorInBST.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            return root
+        else:
```
Comment (nkukarl) [inline inorderSuccessorInBST.py]: No need to do `else`

Comment (peinanteng) [reply inorderSuccessorInBST.py]: sure

File: inorderSuccessorInBST.py (hunk 2 above)
Comment (peinanteng) [reply inorderSuccessorInBST.py]: sure

File: inorderSuccessorInBST.py (hunk 1 above)
Comment (peinanteng) [reply inorderSuccessorInBST.py]: sure

**Pull request merged**